spec/services/exports/export_donations_csv_service_spec.rb

```diff
@@ -30,37 +70,40 @@
         "Comments"
       ] + expected_item_headers
     end
-    let(:expected_item_headers) do
-      item_names = donations.map do |donation|
-        donation.line_items.map(&:item).map(&:name)
-      end.flatten
 
+    let(:total_item_quantities) do
+      template = item_names.index_with(0)
+
+      items_lists.map do |items_list|
+        row = template.dup
+        items_list.each do |(item, quantity)|
+          row[item.name] += quantity
+        end
+        row.values
+      end
```

Comment:
Shouldn't we *know* what this value is? Shouldn't it always be the same number?

Reply:
The test was already generating the expected values programmatically, but I moved the logic up out of the test and into a `let`.

Reply:
Also, we don't know the ordering of the fields, since they are sorted in the export.

Reply:
Hmm... but does the order even matter? When you're doing the expect you can ignore order (e.g. by using `contains_exactly` or even `to_set`).

I'm not going to die on that hill though :) I've noticed that in general the specs in this repo seem to be a lot more dynamic than I feel they should be.

Reply:
Unfortunately the order is pretty important. Since each donation has likely only a subset of the total number of items, there's logic to make sure that the merged set of all items in the collective set of exports is ordered properly and then in each row that each item's totals show up in the right column.

And especially since the output is a bunch of CSV with numeric values, eye-balling it isn't sufficient to notice that, e.g., on row 3 the value in column 14 should actually be in column 16. If each column had a different data type, it would be much easier to notice those kinds of issues at a glance.

Reply:
Okay if I mark this one as resolved?

Reply:
Yep, it's not worth reworking this for a bugfix.